src/app/pages/swap/swap-container.tsx

```diff
@@ -1,66 +1,128 @@
+        x => x.type === 'fungible-token' && alexSDK.getAddressFrom(currency) === x.asset.contractId
+      )?.balance;
+      return {
+        currency,
+        icon: tokenInfo.icon,
+        name: tokenInfo.name,
+        balance: balance ?? createMoney(0, tokenInfo.name, tokenInfo.decimals),
+      };
+    },
+    [allTransferableCryptoAssetBalances]
+  );
 
-  const tempSwapAssetTo: SwapAsset = {
-    balance: createMoney(new BigNumber(0), 'xBTC', 0),
-    icon: XBtcIcon,
-    name: 'Wrapped Bitcoin',
-  };
+  const swappableAssets: SwapAsset[] = useMemo(
+    () => supportedCurrencies.map(getAssetFromAlexCurrency),
+    [getAssetFromAlexCurrency, supportedCurrencies]
+  );
 
-  function onSubmitSwapForReview(values: SwapFormValues) {
-    navigate(RouteUrls.SwapReview, {
-      state: { ...values },
+  const [swapSubmissionData, setSwapSubmissionData] = useState<SwapSubmissionData>();
+  const [slippage, _setSlippage] = useState(0.04);
+
+  async function onSubmitSwapForReview(values: SwapFormValues) {
+    if (values.swapAssetFrom == null || values.swapAssetTo == null) {
+      return;
+    }
+    const [router, lpFee] = await Promise.all([
+      alexSDK.getRouter(values.swapAssetFrom.currency, values.swapAssetTo.currency),
+      alexSDK.getFeeRate(values.swapAssetFrom.currency, values.swapAssetTo.currency),
+    ]);
+    setSwapSubmissionData({
+      swapAmountFrom: values.swapAmountFrom,
+      swapAmountTo: values.swapAmountTo,
+      swapAssetFrom: values.swapAssetFrom,
+      swapAssetTo: values.swapAssetTo,
+      router: router.map(x => getAssetFromAlexCurrency(supportedCurrencies.find(y => y.id === x)!)),
+      liquidityFee: new BigNumber(Number(lpFee)).dividedBy(1e8).toNumber(),
+      slippage,
     });
+    navigate(RouteUrls.SwapReview);
   }
 
-  // TODO: Generate/broadcast transaction > pass real tx data
   function onSubmitSwap() {
+    if (swapSubmissionData == null) {
+      return;
+    }
+    const fromAmount = BigInt(
+      new BigNumber(swapSubmissionData.swapAmountFrom).multipliedBy(1e8).dp(0).toString()
+    );
+    const minToAmount = BigInt(
+      new BigNumber(swapSubmissionData.swapAmountTo)
+        .multipliedBy(1e8)
+        .multipliedBy(1 - slippage)
+        .dp(0)
+        .toString()
+    );
+    const txToBroadcast = alexSDK.runSwap(
+      '', // TODO: current user's stxAddress
+      swapSubmissionData.swapAssetFrom.currency,
+      swapSubmissionData.swapAssetTo.currency,
+      fromAmount,
+      minToAmount,
+      swapSubmissionData.router.map(x => x.currency)
+    );
+    // TODO: broadcast the tx
+    console.log(txToBroadcast);
     navigate(RouteUrls.SwapSummary);
   }
 
+  async function fetchToAmount(
+    from: SwapAsset,
+    to: SwapAsset,
+    fromAmount: string
+  ): Promise<string> {
+    const result = await alexSDK.getAmountTo(
+      from.currency,
+      BigInt(new BigNumber(fromAmount).multipliedBy(1e8).dp(0).toString()),
```

Comment:
not sure what `1e8` is, can we make a named variable to describe it?

Reply:
Hi @kyranjamie we use 8-digit fixed notation to represent decimals, hence multiplying a float by 1e8 and casting to `BigInt`.

Reply:
Got it, would prefer to avoid magic numbers though

```ts
const oneHundredMillion = 100_000_000;
```

We also have a `stxToMicroStx` helper too